src/Microsoft.VisualStudio.Debugger.Lua/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.Debugger.Interop;
+using Microsoft.VisualStudio.Debugger.Lua;
+using System.IO;
+
+namespace Microsoft.VisualStudio.Debugger.Lua
+{
+    // This class represents a pending breakpoint which is an abstract representation of a breakpoint before it is bound.
+    // When a user creates a new breakpoint, the pending breakpoint is created and is later bound. The bound breakpoints
+    // become children of the pending breakpoint.
+    class AD7PendingBreakpoint : IDebugPendingBreakpoint2
+    {
+        private IDebugBreakpointRequest2 m_pBPRequest;
+        private BP_REQUEST_INFO m_bpRequestInfo;
+        private AD7Engine m_engine;
+        private BreakpointManager m_bpManager;
+
+        private System.Collections.Generic.List<AD7BoundBreakpoint> m_boundBreakpoints;
+
+        private bool m_enabled;
+        private bool m_deleted;
+
+        public AD7PendingBreakpoint(IDebugBreakpointRequest2 pBPRequest, AD7Engine engine, BreakpointManager bpManager)
+        {
+            m_pBPRequest = pBPRequest;
+            BP_REQUEST_INFO[] requestInfo = new BP_REQUEST_INFO[1];
+            EngineUtils.CheckOk(m_pBPRequest.GetRequestInfo(enum_BPREQI_FIELDS.BPREQI_BPLOCATION, requestInfo));
+            m_bpRequestInfo = requestInfo[0];
+
+            m_engine = engine;
+            m_bpManager = bpManager;
+            m_boundBreakpoints = new System.Collections.Generic.List<AD7BoundBreakpoint>();
+
+            m_enabled = true;
+            m_deleted = false;
+        }
+
+        public int Bind()
+        {
+            // BP_REQUEST_INFO[] requestInfo = new BP_REQUEST_INFO[1];
+
+            IDebugDocumentPosition2 docPosition = (IDebugDocumentPosition2)(Marshal.GetObjectForIUnknown(m_bpRequestInfo.bpLocation.unionmember2));
+
+            string filename;
+            docPosition.GetFileName(out filename);
```

Comment:
It would surprise me if this didn't return an HR. Probably worth doing EngineUtils.CheckOk if it does